### DISCUSSION TITLE
Is modulepreload polyfilled in vike?

### DISCUSSION THREAD

**Discussion by shishkin17** [vikejs/vike#1446] [Discussion] [answered]
I found that `modulepreload` is polyfilled in vite by default: https://vitejs.dev/config/build-options#build-modulepreload.
But in vike build I can't find it in chunks. Should it be added manually if needed?

**Comment by brillout** [comment]
Same thing than the `?.`: the last time I checked it seemed to be safe to use it.

No user has complained about it so far.

**Reply by shishkin17** [reply]
Its support differs much from es6 modules: https://caniuse.com/?search=modulepreload, especially for safari - it starts only from 17.

**Reply by shishkin17** [reply]
If a browser doesn't support `modulepreload`, these modules will still be loaded, just later, when they are discovered in the entry bundle, right?

**Reply by brillout** [reply, marked as answer]
In general, about browser compatibility, I think the best way to approach it is to wait until a user using a website based on Vike complains that the website doesn't work. It could very well be that Vike isn't the bottleneck but other libraries (e.g. Vue uses ES proxy).

Because, right now, we're opening a can of worms for a questionable added value.

Let's wait and see whether Vike is being a good citizen or not.

That said, if you see quick wins, PRs welcome.

**Reply by brillout** [reply]
Thanks for sponsoring btw. 💚

**Reply by shishkin17** [reply]
I agree, as soon as a site loads without support of `modulepreload`, there is not much added value.
